tests/test_detector.py

```diff
@@ -52,6 +52,6 @@ def test_get_best_opportunity_returns_correct_triplet_with_multiple_tickers():
         ShortTicker(symbol=symbols.Symbol('BTC/TUSD'), last_price=32500),
     ]
     best_opportunity, best_profit = get_best_opportunity(tickers)
-    assert len(best_opportunity) == 3
-    assert best_profit == 5.526315789473684
+    assert len(best_opportunity) >= 3  # Handling cycles with more than 3 tickers
+    assert best_profit == 4.775
```

Comment:
Can you please keep the old test to ensure nothing is broken with the new code and add a new one with more than 3 cycles?

Reply:
I will write separate tests for get best opportunities (new method) vs get best triangular opportunities (old method) then.